src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -600,15 +583,13 @@ protected void finalize() {
     protected final Object poolSizeChangeLock;
     protected final Object workersLock;
     protected final Deque<Worker> availableWorkers;        // is locked around workersLock
-    protected long lastHighDelayMillis;   // is locked around workersLock
-    private int waitingForWorkerCount;  // is locked around workersLock
-    private int currentPoolSize;  // is locked around workersLock
+    protected int waitingForWorkerCount;  // is locked around workersLock
+    protected int currentPoolSize;  // is locked around workersLock
     private final AtomicBoolean shutdownStarted;
     private volatile boolean shutdownFinishing; // once true, never goes to false
-    private volatile long maxWaitForLowPriorityInMs;
```

Comment:
Max wait is now handled in the queue handler, since it is determining what task should be run.  The worker pool only stores workers, and provides them as they are available.  We get the worker before we find out which task to run.